opm/simulators/wells/MultisegmentWell_impl.hpp

```diff
@@ -1598,6 +1598,15 @@ namespace Opm
             // considering the contributions from the inlet segments
             {
                 for (const int inlet : this->segment_inlets_[seg]) {
+
+                    // don't include contribution from shut valves.
+                    if(this->segmentSet()[inlet].segmentType() == Segment::SegmentType::VALVE) {
```

Comment:
Maybe instead of ignoring the contribution from the `SHUT` segment, we should make sure the rate from that segment is `zero`?

Reply:
The crossflow part seems to works now. So the only issue with the current solution is that the output of the segment rate of the closed valve is  different from zero. Setting it to zero leads to a singular system, with the solution propose here we let it be different from zero to avoid singular system, but explicitly exclude the rate from impacting the other segments.

Reply:
I tried to enforce through the primary_variables_[seg][GTotal], but the convergence is really bad. Will think about it more.